packages/grafana-ui/src/components/Combobox/ComboboxList.tsx

```diff
@@ -0,0 +1,153 @@
+import { cx } from '@emotion/css';
+import { useVirtualizer } from '@tanstack/react-virtual';
+import type { UseComboboxPropGetters } from 'downshift';
+import { useCallback } from 'react';
+
+import { useStyles2 } from '../../themes';
+import { Checkbox } from '../Forms/Checkbox';
+import { ScrollContainer } from '../ScrollContainer/ScrollContainer';
+
+import { NotFoundError } from './MessageRows';
+import { getComboboxStyles, MENU_OPTION_HEIGHT, MENU_OPTION_HEIGHT_DESCRIPTION } from './getComboboxStyles';
+import { ALL_OPTION_VALUE, ComboboxOption } from './types';
+import { isNewGroup } from './utils';
+
+export const VIRTUAL_OVERSCAN_ITEMS = 4;
+
+interface ComboboxListProps<T extends string | number> {
+  options: Array<ComboboxOption<T>>;
+  highlightedIndex: number | null;
+  selectedItems?: Array<ComboboxOption<T>>;
+  scrollRef: React.RefObject<HTMLDivElement>;
+  getItemProps: UseComboboxPropGetters<ComboboxOption<T>>['getItemProps'];
```

Comment:
Just a curiosity question, how do you decide when to reference a type like this vs creating a new one? I am never sure what makes more sense, like if I just pass props to other components I am tempted to just ref all the types so they change in at the same time, but that seems like an internal implementation thing. At the same time recreating more complex type is a pain :D

Reply:
> At the same time recreating more complex type is a pain :D

:) this one.

I don't actually really care about the specific type of this, but just that it matches the type from Downshift (where this comes from).